lib/Target/ExportSystemC/Patterns/SystemCEmissionPatterns.cpp

```diff
@@ -91,6 +91,36 @@ struct SignalReadEmitter : OpEmissionPattern<SignalReadOp> {
   }
 };
 
+/// Emit a systemc.cpp.new operation.
+struct NewEmitter : OpEmissionPattern<NewOp> {
+  using OpEmissionPattern::OpEmissionPattern;
+
+  MatchResult matchInlinable(Value value) override {
+    if (value.getDefiningOp<NewOp>())
+      return Precedence::NEW;
+    return {};
+  }
+
+  void emitInlined(Value value, EmissionPrinter &p) override {
+    p << "new ";
+    p.emitType(value.getType().cast<mlir::emitc::PointerType>().getPointee());
+
+    auto newOp = value.getDefiningOp<NewOp>();
+    if (!newOp.getArgs().empty()) {
+      p << "(";
+      bool first = true;
+      for (auto arg : newOp.getArgs()) {
+        if (!first)
+          p << ", ";
```

Comment:
`llvm::interleave_comma` or `llvm::interleave` may help avoid the need for the `bool first` tracking.